File: backend/src/main/java/com/shaderock/backend/organization/supplier/model/entity/Supplier.java
package com.shaderock.backend.organization.supplier.model.entity;

import com.shaderock.backend.order.model.menu.Menu;
import com.shaderock.backend.organization.company.model.entity.Company;
import com.shaderock.backend.organization.model.Organization;
import com.shaderock.backend.organization.supplier.preference.model.entity.SupplierPreferenceConfig;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.net.URI;
import java.util.Set;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
public class Supplier extends Organization {
  @Column(nullable = false)
  private URI websiteUrl;
  @Column(nullable = false)
  private URI menuUrl;

  @JoinTable(name = "lunch_subscriptions",
          joinColumns = @JoinColumn(name = "supplier_id"),
          inverseJoinColumns = @JoinColumn(name = "company_id"))
  @ManyToMany(fetch = FetchType.LAZY)
  private Set<Company> subscribers;

  @JoinTable(name = "lunch_subscriptions_requests",
          joinColumns = @JoinColumn(name = "supplier_id"),
          inverseJoinColumns = @JoinColumn(name = "company_id"))
  @ManyToMany(fetch = FetchType.LAZY)
  private Set<Company> subscriptionsRequests;

  @OneToOne(mappedBy = "supplier", fetch = FetchType.LAZY)
  private Menu menu;

  @OneToOne(mappedBy = "supplier", fetch = FetchType.LAZY)
  private SupplierPreferenceConfig preferences;
}

File: frontend/src/models/Supplier.ts
export class Supplier {
  public id: string | null
  public organizationDetailsId: string | null
  public websiteUrl: string
  public menuUrl: string
  public isPublic: boolean | null
  public preferencesId: string | null

  constructor(id: string | null, organizationDetailsId: string | null, websiteUrl: string, menuUrl: string, isPublic: boolean | null, preferencesId: string | null) {
    this.id = id;
    this.organizationDetailsId = organizationDetailsId;
    this.websiteUrl = websiteUrl;
    this.menuUrl = menuUrl;
    this.isPublic = isPublic;
    this.preferencesId = preferencesId;
  }
}
